cmd/actuary/dockerhost_test.go

```diff
@@ -0,0 +1,425 @@
+package actuary
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/docker/docker/api"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"net/http"          //Package http provides HTTP client and server implementations.
+	"net/http/httptest" //Package httptest provides utilities for HTTP testing.
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+)
+
+// variables for tests, helper functions
+// Rest of test files use the following functions/variables
+// Group together the api call and the expected object (in bytes)
+type callPairing struct {
+	call string
+	obj  []byte
+}
+
+type imageList struct {
+	images []types.ImageSummary
+}
+
+type typeContainerList struct {
+	typeContainers []types.Container
+}
+
+// For testing functions that require a specific number of images
+func (list imageList) populateImageList(size int) imageList {
+	list.images = nil
+	var img types.ImageSummary
+	for i := 0; i < size; i++ {
+		img = types.ImageSummary{ID: strconv.Itoa(i)}
+		list.images = append(list.images, img)
+	}
+	return list
+}
+
+// For testing functions that require a specific number of containers
+func (list typeContainerList) populateContainerList(size int) typeContainerList {
+	list.typeContainers = nil
+	var c types.Container
+	for i := 0; i < size; i++ {
+		c = types.Container{ID: strconv.Itoa(i)}
+		list.typeContainers = append(list.typeContainers, c)
+	}
+	return list
+}
+
+func NewTestTarget(proc []string) (a *Target, err error) {
+	target, err := NewTarget()
+	target.ProcFunc = func(procname string) (cmd []string, err error) {
+		err = nil
+		cmd = proc
+		return
+	}
+	target.CertPath = func(procname string, tlsOpt string) (val string) {
+		val = etcDocker
+		return
+	}
+	a = &target
+	return
+}
+
+func (target *Target) testServer(t *testing.T, pairings ...callPairing) (server *httptest.Server) {
+	var err error
+	mux := http.NewServeMux()
+	for _, pair := range pairings {
+		mux.HandleFunc(
+			fmt.Sprintf("/v1.31%s", pair.call),
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
https://stackoverflow.com/questions/15407719/in-gos-http-package-how-do-i-get-the-query-string-on-a-post-request/15408779#15408779